aioauth/storage.py

```diff
@@ -106,13 +106,14 @@ async def get_id_token(
         scope: str,
         response_type: ResponseType,
         redirect_uri: str,
-        nonce: str,
+        nonce: Optional[str],
```

Comment:
`nonce` is optional since a `token` request would not include a `nonce`. That said, a `nonce` would probably be available as part of data stored with the authorization code. But it kind of seems like we'd leave it to the custom implementation of `aioauth.storage.Storage` to fetch the `nonce` from the authorization code data and include it with the `id_token`. Maybe it's preferable if the library handles the fetching of the `nonce` from authorization code data? I'm open to other ideas.

Reply:
It seems like the OIDC spec would want the `nonce` in an `id_token` for a `token` response, but it's not very well spelled out anywhere.

The example token response here does include `nonce` : https://openid.net/specs/openid-connect-core-1_0.html#TokenResponse

The token endpoint documentation just points back to the hybrid flow token response: https://openid.net/specs/openid-connect-core-1_0.html#HybridTokenResponse

So, not sure what's right here, but I lean toward the `id_token` having a `nonce` when getting `id_token` via the token endpoint.